chainstate/tx-verifier/src/transaction_verifier/hierarchy.rs

```diff
@@ -178,18 +168,18 @@ impl<'a, S: TransactionVerifierStorageMut> TransactionVerifierStorageMut
     fn set_undo_data(
         &mut self,
         id: Id<Block>,
-        undo: &BlockUndo,
+        new_undo: &BlockUndo,
     ) -> Result<(), TransactionVerifierStorageError> {
-        match self.utxo_block_undo.entry(id) {
-            Entry::Occupied(_) => Err(TransactionVerifierStorageError::DuplicateBlockUndo(id)),
-            Entry::Vacant(entry) => {
-                entry.insert(BlockUndoEntry {
-                    undo: undo.clone(),
-                    is_fresh: true,
-                });
-                Ok(())
-            }
-        }
+        self.utxo_block_undo
+            .entry(id)
+            .and_modify(|cached_undo| {
+                cached_undo.undo.append(new_undo.clone());
```

Comment:
please pay attention to this line. I have doubts whether it is ok to call `append` to merge `BlockUndo`, because the function is called `set_undo_data` and there is no guarantee that info in `new_undo` actually corresponds to txs with later indexes.  Though this works for our code. What do you think?